test/test_log.py

```diff
@@ -162,21 +164,34 @@ def test_5(self):
         self.assertTrue(name_filter(record))
         self.assertEqual(record.name, '__init__/module')
 
-    def test_plugin_path_long(self):
+    def test_plugin_path_long_1(self):
         DebugOpt.PLUGIN_FULLPATH.enabled = True
-        record = FakeRecord(name=None, pathname='/path1/path2/plugins/path3/plugins/plugin.zip')
+        record = FakeRecord(name=None, pathname='/user/picard/plugins/plugin.zip')
         self.assertTrue(name_filter(record))
-        self.assertEqual(record.name, 'plugins/path3/plugins/plugin')
+        self.assertEqual(record.name, 'user/picard/plugins/plugin')
+
+    def test_plugin_path_long_2(self):
+        DebugOpt.PLUGIN_FULLPATH.enabled = True
+        record = FakeRecord(name=None, pathname='/user/picard/plugins/plugin.zip/plugin')
+        self.assertTrue(name_filter(record))
+        self.assertEqual(record.name, 'user/picard/plugins/plugin.zip/plugin')
+
+    def test_plugin_path_short_1(self):
+        DebugOpt.PLUGIN_FULLPATH.enabled = False
+        record = FakeRecord(name=None, pathname='/user/picard/plugins/plugin.zip')
+        self.assertTrue(name_filter(record))
+        self.assertEqual(record.name, 'plugins/plugin')
 
-    def test_plugin_path_short(self):
+    def test_plugin_path_short_2(self):
         DebugOpt.PLUGIN_FULLPATH.enabled = False
-        record = FakeRecord(name=None, pathname='/path1/path2/plugins/path3/plugins/plugin.zip')
+        record = FakeRecord(name=None, pathname='/user/picard/plugins/plugin.zip/plugin')
```

Comment:
It would be better to use `'/user/picard/plugins/plugin.zip/xxx' here because we don't know which `plugin` was kept in the result.